src/server/terrain/simple_structures/SbbGen.zig

```diff
@@ -57,19 +76,27 @@ fn placeSbb(self: *SbbGen, structure: *const sbb.StructureBuildingBlock, placeme
 
 	for(rotated.childBlocks) |childBlock| {
 		const child = structure.pickChild(childBlock, seed) orelse continue;
-		placeSbb(self, child, pastePosition + childBlock.pos(), childBlock.direction(), chunk, seed);
+		const childRotation = switch(childBlock.direction()) {
+			.dirDown, .dirUp => switch(child.rotation) {
```

Comment:
This should be a member function in my opinion.

Reply:
Anyway, extracted to `Rotation`